### PR TITLE
Revert erroneous change which drops user events on some completions.

### PR DESCRIPTION
**Pull request by rabbah**
This fixes a bug introduced in a previous PR which drops user events on completion messages that do not carry a result (which is needed for user event generation).

Best to review this ignoring white space.

It appears there are no tests that caught the bug and so some tests are needed. 
@sven-lange-last perhaps to verify the fix you can suggest a test to add 🙏 

## Related issue and scope

- [ ] I opened an issue to propose and discuss this change (#????)

## My changes affect the following components

- [ ] API
- [ ] Controller
- [ ] Message Bus (e.g., Kafka)
- [ ] Loadbalancer
- [x] Invoker
- [ ] Intrinsic actions (e.g., sequences, conductors)
- [ ] Data stores (e.g., CouchDB)
- [ ] Tests
- [ ] Deployment
- [ ] CLI
- [ ] General tooling
- [ ] Documentation

## Types of changes

- [x] Bug fix (generally a non-breaking change which closes an issue).
- [ ] Enhancement or new feature (adds new functionality).
- [ ] Breaking change (a bug fix or enhancement which changes existing behavior).

## Checklist:

- [x] I signed an [Apache CLA](https://github.com/apache/openwhisk/blob/master/CONTRIBUTING.md).
- [x] I reviewed the [style guides](https://github.com/apache/openwhisk/wiki/Contributing:-Git-guidelines#code-readiness) and followed the recommendations (Travis CI will check :).
- [ ] I added tests to cover my changes.
- [ ] My changes require further changes to the documentation.
- [ ] I updated the documentation where necessary.

### CODE REVIEW

**Comment by chetanmeh** [comment]
Currently User Event feature does not have full coverage as it's disabled by default. 

With support for embedded Kafka in Standalone we can now add an integration test where Invoker is started with user event enabled like being done in [Activation Persister IT](https://github.com/chetanmeh/incubator-openwhisk/blob/persister/tests/src/test/scala/org/apache/openwhisk/core/database/persister/ActivationPersisterITTests.scala) and then assert if messages are indeed being sent.

**Comment by sven-lange-last** [comment]
PG 1 / 25

**Comment by chetanmeh** [comment]
I have pushed a IT test [here][1] which current fails on master but passes with this PR. If it seems fine I can push that to current PR branch

[1]: https://github.com/rabbah/openwhisk/compare/userev...chetanmeh:userev-test

**Comment by rabbah** [comment]
Thanks @chetanmeh - I'll defer to you since it's more work for you.

**Comment by rabbah** [comment]
Thanks @chetanmeh! 

Note to @sven-lange-last when merging this PR please do not squash to preserve Chetan's commit.

**Comment by sven-lange-last** [comment]
I ran following `UserEventTests` in our build / test pipeline:

https://github.com/apache/openwhisk/blob/master/tests/src/test/scala/org/apache/openwhisk/common/UserEventTests.scala

The test failed once (PG 1 / 25) because it only waits for the user event message to arrive on Kafka for 10 seconds. So from my perspective, we should increase the maximum wait time to make the test a little more robust. A longer max poll time should not harm in most cases because the `consumer.peek()` will return as soon as messages could be received.

The next test run of `UserEventTests` (PG 1 / 26) succeeded with this PR. It always fails with current master.

In other words: this PR fixes the missing user events.

**Comment by chetanmeh** [comment]
Changes look fine and can now be merged

**Comment by sven-lange-last** [comment]
@steffenrost plans to improve the current `UserEventTests` as well as trying to a unit tests for user event sending.
